python/python/lance/dataset.py

```diff
@@ -823,6 +826,11 @@ def create_index(
         self._ds.create_index(column, index_type, name, metric, kwargs)
         return LanceDataset(self.uri)
 
+    def optimize_indices(self, **kwargs) -> LanceDataset:
+        """Optimize indices."""
```

Comment:
```suggestion
        """
        Optimizes index performance

        As new data arrives it is not added to existing indexes automatically.
        When searching we need to perform an indexed search of the old data plus
        an expensive unindexed search on the new data.  As the amount of new
        unindexed data grows this can have an impact on search latency.

        This function will add the new data to existing indexes, restoring the
        performance.  This function does not retrain the index, it only assigns
        the new data to existing partitions.  This means an update is much quicker
        than retraining the entire index but may have less accuracy (especially
        if the new data exhibits new patterns, concepts, or trends)
        """
```

Here's the wording I came up with in my test PR.  Feel free to modify as you want.  It's just a suggestion.  But I think users will want more details.

Reply:
Got it. Lemme copy this.